crates/query-engine/translation/src/translation/query/filtering.rs

```diff
@@ -913,3 +913,171 @@ fn underlying_type_name(typ: database::Type) -> models::TypeName {
+    let mut all_are_eq_comparisons = true;
+
+    for expr in expressions {
+        match expr {
+            models::Expression::BinaryComparisonOperator {
+                column,
+                operator,
+                value,
+            } => {
+                // Check if this is an equality operator
+                let left_typ = get_comparison_target_type(env, current_table_scope, column)?;
+                let op = env.lookup_comparison_operator(&left_typ, operator)?;
+
+                if op.operator_kind != metadata::OperatorKind::Equal {
+                    all_are_eq_comparisons = false;
+                    break;
+                }
+
+                // Only handle simple column comparisons for now
+                if let models::ComparisonTarget::Column {
+                    name, field_path, ..
+                } = column
+                {
+                    if field_path.as_ref().is_none_or(|fp| fp.is_empty()) {
+                        // Group by column name
+                        let column_group = column_groups.entry(name).or_default();
+                        column_group.comparison_values.push(value);
+                        column_group.expressions.push(expr);
+                    } else {
+                        all_are_eq_comparisons = false;
+                        break;
+                    }
+                } else {
+                    all_are_eq_comparisons = false;
+                    break;
+                }
+            }
+            _ => {
+                all_are_eq_comparisons = false;
+                break;
+            }
+        }
+    }
+
+    if !all_are_eq_comparisons {
+        return Ok(None);
+    }
+
+    let mut expressions = vec![];
+
+    for (
+        column_name,
+        ColumnGroup {
+            comparison_values,
+            expressions: original_expressions,
+        },
+    ) in column_groups
+    {
+        // Check if all values are scalar values (not column references)
+        if comparison_values.len() >= 2
+            && comparison_values
+                .iter()
+                .all(|v| matches!(v, models::ComparisonValue::Scalar { .. }))
+        {
+            if let Some(in_expr) = create_in_expression(column_name, comparison_values) {
+                expressions.push(in_expr);
+            } else {
+                expressions.extend(
+                    original_expressions
+                        .into_iter()
+                        .cloned()
+                        .collect::<Vec<_>>(),
+                )
+            }
+            // try and construct an `in` and push to expressions
+        } else {
+            // push the regular `or`
+            expressions.extend(
+                original_expressions
+                    .into_iter()
+                    .cloned()
+                    .collect::<Vec<_>>(),
+            )
+        }
+    }
+
+    Ok(Some(expressions))
+}
+
+fn create_in_expression(
```

Comment:
Previously Augment did the `models::Expression` -> `sql::ast::Expression` transformation here too, which was messy and confusing. Good idea though, generally.